main/src/main/scala/sbt/Main.scala

```diff
@@ -940,14 +940,28 @@ object BuiltinCommands {
     state.remainingCommands exists (_.commandLine == TemplateCommand)
 
   private def writeSbtVersion(state: State) =
-    if (SysProp.genBuildProps && !intendsToInvokeNew(state))
+    if (SysProp.genBuildProps && !intendsToInvokeNew(state)) {
       writeSbtVersionUnconditionally(state)
+    }
+
+  private def checkRoot(state: State): Unit =
+    if (SysProp.allowRoot) ()
+    else {
+      val baseDir = state.baseDir
+      if (baseDir.getAbsolutePath == "/") {
```

Comment:
Is there an equivalent check for windows? You probably don't want to start sbt in `C:\\` either. Perhaps this should be something like
```scala
  val basePath = state.baseDir.toPath.toAbsolutePath
  val roots = FileSystems.getDefault.getRootDirectories.asScala.toSet
  if (roots.contains(basePath)) { ...
```

Reply:
We probably want to prevent `C:\` but `D:\` should be ok?

Reply:
I don't know enough about windows to answer that. I guess you could probably take the head of `getRootDirectories` or filter for `C:\` on windows.

Reply:
Addressed in 563bcb9